src/main/java/gregtech/common/pipelike/PipeCollectorWalker.java

```diff
@@ -0,0 +1,67 @@
+package gregtech.common.pipelike;
+
+import gregtech.api.pipenet.PipeNetWalker;
+import gregtech.api.pipenet.tile.IPipeTile;
+
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class PipeCollectorWalker<T extends IPipeTile<?, ?>> extends PipeNetWalker<T> {
+
+    @NotNull
+    public static List<IPipeTile<?, ?>> collectPipeNet(World world, BlockPos sourcePipe, IPipeTile<?, ?> pipe,
+                                                       int limit) {
+        PipeCollectorWalker<? extends IPipeTile<?, ?>> walker = (PipeCollectorWalker<? extends IPipeTile<?, ?>>) new PipeCollectorWalker<>(
+                world, sourcePipe, 0, pipe.getClass());
+        walker.traversePipeNet(limit);
```

Comment:
Note that `limit != max_pipe_walked` most of the time. You should rather check the collect pipe amount and call `stop()` (or `getRoot().stop()` idr). But that might mess with pipes wich are already painted. Instead of collecting the pipes you could hava a `Consumer<IPipeTile>` in the walker which paints the pipe directly and check spray can durability each time.